carts/serializers.py

```diff
@@ -0,0 +1,71 @@
+"""
+Serializers for carts models
+"""
+from django.utils.translation import gettext_lazy as _
+from rest_framework import serializers
+
+from carts.models import Cart, CartItem
+
+
+class CartSerializer(serializers.ModelSerializer):
+    """
+    Cart model serializer
+    """
+
+    def create(self, validated_data):
+        validated_data["created_by"] = validated_data["user"]
+        return Cart.objects.create(**validated_data)
+
+    class Meta:
+        """
+        Tells the fields to include in parsed/json object
+        """
+
+        # pylint: disable=protected-access, no-member
+        model = Cart
+        fields = "__all__"
+
+
+class CartItemSerializer(serializers.ModelSerializer):
+    """
+    CartItem model serializer
+    """
+
+    def create(self, validated_data):
+        """
+        Creates a CartItem
+        Args:
+            validated_data(dict): Value containing verified data
+        Returns:
+            (CartItem): Value containing created CartItem instance
+        """
+        validated_data["created_by"] = validated_data["cart"].user
+        filter_item = CartItem.objects.filter(
+            product=validated_data["product"], cart=validated_data["cart"]
+        )
```

Comment:
Can we use a reverse relation e.g. just getting the cart and get the items using that cart object?

Reply:
Yes, Arslan bhai we can do that but, we don't want all the `cart_items` we just want if the same product already exists in cart and if it does we just change its quantity rather than creating a new `cart_item`. Furthermore, if we use reverse relation we would have to traverse all the `cart_items` and check if the product exists or not. I think this is not a good approach. Please let me know if this is not the case

Reply:
You are currently running a filter against all the cart items in the table, what will happen in the reverse relation of cartitem_set is that you will still be running the filter on the queryset returned by that and not the whole table. Did you get my point?

Reply:
Yes, I get it now. Thanks for the clarification